pipx/commands.py

```diff
@@ -327,6 +335,71 @@ def install(
         venv.remove_venv()
         raise
 
+    # if all is well, write out pipxrc file
+    package_or_url_pipxrc = _abs_path_if_local(package_or_url, venv, pip_args)
+    pipxrc = Pipxrc(venv_dir, read=False)
```

Comment:
I find this a bit un-pythonic; what if instead we do:

```
PipxRc()
```

when not reading from file, and then we can have a classmethod

```
PipxRc.from_file()
```

when we do read from a file.

Reply:
This is a perfectly valid way of doing it, but my personal preference would be to leave it as is. There is only one use of instantiating Pipxrc where we don't read from a file.  I would lean to eliminating the boiler-plate `from_file()` for all the other times when we do want to read from file.

Reply:
Wouldn't the from_file be call the default, set the values as read from file. Passing arguments is something that would come up a lot more if you start adding tests.

Reply:
Yes, `from_file()` would be the default.  What I was trying to say was, I didn't want the default, i.e. reading from the file, to be the thing that always needs extra code tacked on the end of the instantiation.  That's why I made it an `read` an argument with a `True` default, because `read=False` only happens exactly once in the whole codebase.

Reply:
Reading the content from a file is extra job to be performed, as opposed from just passing in the arguments from memory, or leaving it at default values. If anything for me makes it more readable and explicit what's happening than a magic flag that causes bifurcation in the code.

```
PipxRc(package_or_url=whgat) - in-memory/default
PipxRc.from_file()            - reads from file the arguments  
```

This would be in-line with how the standard library is designed; 

https://docs.python.org/3/library/datetime.html#datetime.date.fromtimestamp
rather than datetime.date(is_timestamp=True)

Reply:
I don't see how `PipxRc.from_file()` is any less magic than `Pipxrc(venv_dir, read=False)`. If anything, I find static methods and class methods more confusing ([GVR himself](https://www.webucator.com/blog/2016/05/when-to-use-static-methods-in-python-never/) even says static methods were a mistake but it was too late to remove them). I appreciate the focus on clean code, but I am fine with what is currently in this PR. Would `Pipxrc(venv_dir, use_file_if_present=False)` or something similar make you more comfortable?

Reply:
I do still like the idea of making the pipxrc (or whatever we end up naming it) object an instance variable on the `Venv` class. 

Something like 
```
venv = Venv(..., read_pipxrc_file=True)
venv.pipxrc  # it is always accessed like this (and again, maybe we rename it from pipxrc) 
```

Reply:
> I guess we'll have to agree to disagree on this and the importance of clean code.

We definitely agree on clean code! Please don't get the impression that I'm just throwing things together and shipping it. 

I hope that is apparent from the current state of pipx with many CI tests (with hope for more), lint tests, manifest tests, types everywhere, coverage report, and code readability. When given the choice, I almost always opt for developer sanity and code readability over supporting corner cases that will make the package slightly more useful but twice as confusing to read the code. That is the reason I chose Python3.6+ and stuck to it.

> This is the same reason I've been suggesting for adding tests; so people can modify the code without fearing to break current behaviour; and as such the project becomes easier to maintain.

Totally agree. I suggested this too, and so did @itsayellow; he said that up front and is just wanted to make sure we agreed on an implementation before writing tests so he could avoid churn. My approval was probably premature, I'll take it off for now.

What I was trying to say is that although I disagree with your suggestion, I appreciate that you're pushing for clean code, even though we may sometimes disagree on what "clean" means (I don't think `Pipxrc(venv_dir, read=False)` is adding tech debt).

I also remember you mentioning breaking up `commands.py` into separate files since it might be hard for new devs to navigate in one big file. I kind of like the big file, but I can see how separating everything might make it easier so I would not be opposed to that.